src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -122,7 +122,7 @@ public static ClassName bestGuess(String classNameString) {
 
     // Add the package name, like "java.util.concurrent", or "" for no package.
     int p = 0;
-    while (p < classNameString.length() && Util.isLowerCase(classNameString.charAt(p))) {
+    while (p < classNameString.length() && Character.isLowerCase(classNameString.charAt(p))) {
```

Comment:
Fair enough. It's not a reason not to do things properly, though. Just because people _might_ have problems if they use such identifiers doesn't mean that we can just throw them under the bus here.